### PR TITLE
Add autocompletion for namespaces (IPython only)

### PR DESCRIPTION
**Pull request by kysrpex**
This pull request aims to solve issue #591, at least for IPython users. The idea is to modify the `__dir__` method of namespaces to include their ontology entities, so that auto completion is offered. A similar thing could maybe be done for the `namespaces.py` files, I still have to do more research on this.

To Do list to complete the pull request
- [x] Add autocompletion for Ontology Entities inside a namespace.
- [x] Add autocompletion of namespaces (when writing `osp.core.namespaces.[...]`).
- [x] Write unit tests for both cases.

### CODE REVIEW

**Comment by kysrpex** [comment]
Updated the checklist (added autocompletion for namespaces).

**Comment by kysrpex** [comment]
Updated the checklist (added autocompletion for namespaces). This will only work if the user first imports the `namespaces` module, i.e. `import osp.core.namespaces as namespaces`. Then autocompletion will be available for the `namespaces` object. I think it is not possible to autocomplete the statement `from osp.core.namespaces import <...>`.

File: tests/test_namespace.py [hunk 1]
```diff
@@ -515,6 +515,16 @@ def test_get_entity_name(self):
...
 
+    def test_autocompletion_ipython(self):
+        """Checks that all the expected ontology entities are in __dir__.
+
+        The check is done just for the `cuba` namespace.
+        """
+        expected = {'activeRelationship', 'passiveRelationship',
+                    'relationship', 'attribute', 'path', 'Entity', 'File',
+                    'Nothing', 'Wrapper'}
+        self.assertSetEqual(set(cuba.__dir__()) & expected, expected)
```
Comment (pablo-de-andres) [inline tests/test_namespace.py:526]: Here I would call `dir(cuba)`, which is what the magic function is acting for.

File: osp/core/ontology/namespace.py [hunk 2]
```diff
@@ -32,6 +32,17 @@ def __init__(self, name, namespace_registry, iri):
...
+    def __dir__(self):
+        """Attributes available for the OntologyNamespace class.
+
+        Returns:
+            Iterable: the available attributes, which include the methods and
+                      the ontology entities in the namespace.
+        """
+        entity_autocompletion = self._iter_labels() \
+            if self._reference_by_label else self._iter_suffixes()
+        return itertools.chain(super().__dir__(), entity_autocompletion)
```
Comment (pablo-de-andres) [inline osp/core/ontology/namespace.py:44]: I would call `dir(super())`

File: osp/core/ontology/namespace.py [hunk 3]
```diff
@@ -254,20 +265,46 @@ def _get_from_label(self, label, lang=None, case_sensitive=False):
...
-        :return: An iterator over the entities.
-        :rtype: Iterator[OntologyEntity]
+        :return: An iterator over the entity IRIs.
+        :rtype: Iterator[rdflib.URIRef]
         """
         types = [rdflib.OWL.DatatypeProperty,
                  rdflib.OWL.ObjectProperty,
                  rdflib.OWL.Class]
-        return (self._namespace_registry.from_iri(s)
+        return (s
```
Comment (pablo-de-andres) [inline osp/core/ontology/namespace.py:277]: I don't think this is an Iterator, is it?

File: osp/core/ontology/namespace.py [hunk 4]
```diff
@@ -254,20 +265,46 @@ def _get_from_label(self, label, lang=None, case_sensitive=False):
...
                 for t in types
                 for s, _, _ in self._graph.triples((None, rdflib.RDF.type, t))
                 if s in self)
 
+    def __iter__(self):
```
Comment (pablo-de-andres) [inline osp/core/ontology/namespace.py:282]: And this doesn't look like an Iterator either

File: tests/test_namespace.py (hunk 1 above)
Comment (kysrpex) [reply tests/test_namespace.py]: Done, thanks!

File: osp/core/ontology/namespace.py (hunk 2 above)
Comment (kysrpex) [reply osp/core/ontology/namespace.py]: Done.

**Review by pablo-de-andres — APPROVED** [review]

**Pull request merged**